backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java

```diff
@@ -413,4 +420,27 @@ public void updateOrderPaySchedules(@NonNull final I_C_Invoice invoice)
 			orderBL.syncDateInvoicedFromInvoice(orderId, invoice);
 		}
 	}
+
+	@DocValidate(timings = ModelValidator.TIMING_BEFORE_COMPLETE)
+	public void copyProjectIdFromLinesIfApplicable(final I_C_Invoice invoice)
+	{
+		final Set<ProjectId> projectIds = invoiceDAO.retrieveLines(invoice)
+				.stream()
+				.map(line -> ProjectId.ofRepoIdOrNull(line.getC_Project_ID()))
+				.filter(Objects::nonNull)
+				.collect(Collectors.toSet());
+		if (projectIds.size() == 1)
+		{
+			invoice.setC_Project_ID(projectIds.iterator().next().getRepoId());
+		}
+		else
+		{
+			if (projectIds.size() > 1)
+			{
+				log.debug("Invoice {} has multiple projects {}, not setting header project",
+						invoice.getC_Invoice_ID(), projectIds);
```

Comment:
minor: avoid method calls when logging, e.g. invoice.getC_Invoice_ID(). Use instead invoice.
reason: if debug logging is not enabled the impact shall be ZERO.